src/main/java/io/pinecone/commons/IndexInterface.java

```diff
@@ -266,67 +377,323 @@ default DescribeIndexStatsRequest validateDescribeIndexStatsRequest(Struct filte
+     * Retrieves the IDs of the most similar items, along with their similarity scores, and optionally, their values and metadata.
+     *
+     * @param topK The number of top similar vectors to retrieve.
+     * @param id The unique ID of a vector to be used as a query vector.
+     * @param includeValues Flag indicating whether to include the vector values in the query results.
+     * @param includeMetadata Flag indicating whether to include the vector metadata in the query results.
+     * @return A generic type {@code U} representing the query results, including optional values and metadata based on parameters.
+     */
     U queryByVectorId(int topK, String id, boolean includeValues, boolean includeMetadata);
 
+    /**
+     * Queries a specified namespace using a vector ID. Retrieves the IDs of the most similar items, along with their similarity scores.
+     *
+     * @param topK The number of top similar vectors to retrieve.
+     * @param id The unique ID of a vector to be used as a query vector.
+     * @param namespace The namespace within which the query is performed.
+     * @return A generic type {@code U} representing the query results.
+     */
     U queryByVectorId(int topK, String id, String namespace);
 
+    /**
+     * Queries a specified namespace using a vector ID, with options to include values and metadata in the results.
+     * Retrieves the IDs of the most similar items, along with their similarity scores, and optionally, their values and metadata.
+     *
+     * @param topK The number of top similar vectors to retrieve.
+     * @param id The unique ID of a vector to be used as a query vector.
+     * @param namespace The namespace within which the query is performed.
+     * @param includeValues Flag indicating whether to include the vector values in the query results.
+     * @param includeMetadata Flag indicating whether to include the vector metadata in the query results.
+     * @return A generic type {@code U} representing the query results, including optional values and metadata based on parameters.
+     */
     U queryByVectorId(int topK, String id, String namespace, boolean includeValues, boolean includeMetadata);
 
+    /**
+     * Queries a specified namespace using a vector ID and a custom filter. Retrieves the IDs of the most similar items, along with their similarity scores.
+     *
+     * @param topK The number of top similar vectors to retrieve.
+     * @param id The unique ID of a vector to be used as a query vector.
+     * @param namespace The namespace within which the query is performed.
+     * @param filter A Struct defining additional filtering criteria for the query.
+     * @return A generic type {@code U} representing the query results, filtered according to the provided criteria.
+     */
     U queryByVectorId(int topK, String id, String namespace, Struct filter);
 
-    U queryByVectorId(int topK, String id, String namespace, Struct filter, boolean includeValues,
```

Comment:
Why was this removed?

Reply:
It wasn't, I think it's just the formatting. It's down on ln. 489.

Reply:
Ahh okay, sorry